x/mongo/driver/operation/ismaster.go

```diff
@@ -194,11 +194,12 @@ func (im *IsMaster) handshakeCommand(dst []byte, desc description.SelectedServer
 }
 
 // command appends all necessary command fields.
-func (im *IsMaster) command(dst []byte, _ description.SelectedServer) ([]byte, error) {
-	if im.serverAPI != nil {
+func (im *IsMaster) command(dst []byte, desc description.SelectedServer) ([]byte, error) {
+	if im.serverAPI != nil || desc.HelloOK {
 		dst = bsoncore.AppendInt32Element(dst, "hello", 1)
```

Comment:
I think this should include `helloOk` based on the pseudo code in: https://github.com/mongodb/specifications/blob/master/source/server-discovery-and-monitoring/server-monitoring.rst#monitor-thread

Reply:
Ah great catch. Moved the `helloOk` append out of the if check.